lib/rubycritic/generators/xml_report.rb

```diff
@@ -0,0 +1,24 @@
+require 'rubycritic/generators/xml/simple'
+
+module Rubycritic
+  module Generator
+    class XmlReport
+      def initialize(analysed_modules)
+        @analysed_modules = analysed_modules
+      end
+
+      def generate_report
+        report = generator.render
+
+        File.write "#{ Config.root }/report.xml", report
+        print report
+      end
+
+      private
+
+      def generator
+        Xml::Simple.new(@analysed_modules)
```

Comment:
Hi. Sorry for the late response. Long story short, we needed even more changes like picking which analysers to run. This diverges much from the original code. So I think, it would be a better idea to just keep those changes inside our fork. Sorry for wasting your time.